components/plugin-manager/implementation/src/main/kotlin/PluginEvent.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2025 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.components.pluginmanager
+
+import kotlinx.datetime.Instant
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+internal data class PluginEvent(
+    val pluginType: PluginType,
+    val pluginId: String,
+    val version: Long,
+    val payload: PluginEventPayload,
+    val createdAt: Instant,
+    val createdBy: String
+)
+
+/** The base class for all plugin event payloads. */
+@Serializable
+internal sealed class PluginEventPayload
+
+/** The payload for a plugin enabled event. */
+@Serializable
+@SerialName("PluginEnabled")
```

Comment:
Why is this necessary?

Reply:
This makes the serialization more stable, because by default the whole qualified name is used, which means that if the package of the class is changed old entries in the database could not be deserialized anymore.